@here/harp-mapview/lib/MapView.ts

```diff
@@ -1797,19 +1834,51 @@ export class MapView extends THREE.EventDispatcher {
             this.m_forceCameraAspect !== undefined ? this.m_forceCameraAspect : width / height;
         this.setFovOnCamera(this.m_options.fovCalculation!, height);
 
-        const kMinNear = 0.1;
-        const kMultiplier = 50.0;
-        const kFarOffset = 200.0;
-
-        let nearPlane = Math.max(kMinNear, this.m_camera.position.z * 0.1);
-        let farPlane = nearPlane * kMultiplier + kFarOffset;
-
-        if (this.m_options.farPlaneEvaluator !== undefined) {
-            this.camera.getWorldDirection(this.m_tempVector3);
-            const angle = THREE.Math.radToDeg(this.m_tempVector3.angleTo(EYE_INVERSE));
-            const nearFarPlane = this.m_options.farPlaneEvaluator(this, angle, nearPlane, farPlane);
-            nearPlane = Math.max(kMinNear, nearFarPlane.near);
-            farPlane = Math.max(nearPlane + kFarOffset, nearFarPlane.far);
+        let nearPlane: number = 0;
+        let farPlane: number = 0;
+
+        if (this.projection.type === ProjectionType.Spherical) {
+            // near and far plane for a set up where
+            // the camera is looking at the center of the scene.
+            const r = EarthConstants.EQUATORIAL_RADIUS;
+            const d = this.m_camera.position.length();
+            const alpha = Math.asin(r / d);
+            const xaxis = new THREE.Vector3();
+            const yaxis = new THREE.Vector3();
+            const zaxis = new THREE.Vector3();
+            this.m_camera.matrixWorld.extractBasis(xaxis, yaxis, zaxis);
+            const q = new THREE.Quaternion();
+            q.setFromAxisAngle(xaxis, alpha);
+            const fwd = zaxis.clone().negate();
+            const p = this.m_camera.position.clone();
+            p.addScaledVector(fwd.clone().applyQuaternion(q), Math.sqrt(d * d - r * r));
+            farPlane = p
+                .clone()
+                .sub(this.m_camera.position)
+                .dot(fwd);
+            const bias = 2000; // TODO: generalize.
+            nearPlane = Math.max(1, this.projection.groundDistance(this.m_camera.position) - bias);
+        } else {
+            const kMinNear = 0.1;
+            const kMultiplier = 50.0;
+            const kFarOffset = 200.0;
+
+            const groundDistance = this.projection.groundDistance(this.m_camera.position);
+            nearPlane = Math.max(kMinNear, groundDistance * 0.1);
+            farPlane = nearPlane * kMultiplier + kFarOffset;
+
+            if (this.m_options.farPlaneEvaluator !== undefined) {
```

Comment:
Add comment in `MapViewOptions` that `farPlaneEvaluator` is only supported with flat projections.

Reply:
i think farPlaneEvaluator like many of the options of MapViewOptions should go away. They are a bit too low level

Reply:
I can try to make the far plane evaluator work with globe as well. I need to refactor a bit the farPlaneEvaluator interface and its usage (e.g. the clamping), I will create a task and make it work in a follow up change.